ytmusicapi/mixins/browsing.py

```diff
@@ -75,41 +98,60 @@ def search(self, query: str, filter: str = None) -> List[Dict]:
                     param2 = 'RAAGAAgACgA'
                 params = param1 + param2 + param3
 
-            body['params'] = params
-
-        response = self._send_request(endpoint, body)
-
-        try:
-            # no results
-            if 'contents' not in response:
-                return search_results
+        type = filter[:-1] if filter else None
 
-            if 'tabbedSearchResultsRenderer' in response['contents']:
-                results = response['contents']['tabbedSearchResultsRenderer']['tabs'][int(
-                    filter == "uploads")]['tabRenderer']['content']
-            else:
-                results = response['contents']
+        if not continuationToken:
+            body = {"query": q, "params": params}
 
-            results = nav(results, SECTION_LIST)
+        if continuationToken:
+            query = {"cToken": continuationToken, "continuation": continuationToken}
 
-            # no results
-            if len(results) == 1 and 'itemSectionRenderer' in results:
-                return search_results
+        response = self._send_request(endpoint, body, query=query)
 
-            for res in results:
-                if 'musicShelfRenderer' in res:
-                    results = res['musicShelfRenderer']['contents']
+        try:
+            results = None
+
+            if "continuationContents" in response:
+                results = response["continuationContents"]
+            elif "contents" in response:
+                if "tabbedSearchResultsRenderer" in response["contents"]:
+                    results = response["contents"]["tabbedSearchResultsRenderer"]["tabs"][int(
+                        filter ==  # noqa: W504
```

Comment:
This looks really odd. Try to put the comparison in one line. There should be no need for noqa

Reply:
Yes, it really does look strange - the automatic formatting tool went crazy and I couldn't work out how to get it to pass without the noqa.  I'll try again.